load-gen/src/main/java/io/smartcat/data/loader/datasource/CSVDataSource.java

```diff
@@ -0,0 +1,75 @@
+package io.smartcat.data.loader.datasource;
+
+import io.smartcat.data.loader.api.DataSource;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+import java.util.Iterator;
+
+/**
+ * Data source that reads data from CSV file.
+ *
+ * @param <T> Type of the data provided by data source implementation.
+ */
+public class CSVDataSource<T> implements DataSource<T>, AutoCloseable {
+
+    private final CSVParser csvParser;
+    private final RowMapper<T> rowMapper;
+
+    private Iterator<CSVRecord> iterator;
+    private boolean closed = false;
+
+    /**
+     * COnstructs CSV data source with specified <code>csvParser</code> and <code>rowMapper</code>.
+     * @param csvParser parser which provides CSV records to this data source.
+     * @param rowMapper mapper which maps CSV records to type {@code <T>}.
+     */
+    public CSVDataSource(CSVParser csvParser, RowMapper<T> rowMapper) {
+        if (csvParser == null) {
+            throw new IllegalArgumentException("CSV parser cannot be null.");
+        }
+        if (rowMapper == null) {
+            throw new IllegalArgumentException("Row mapper cannot be null.");
+        }
+        this.rowMapper = rowMapper;
+        this.csvParser = csvParser;
+        this.iterator = this.csvParser.iterator();
+    }
+
+    @Override
+    public boolean hasNext(long time) {
+        if (closed) {
+            throw new IllegalStateException("Data source is already closed.");
+        }
+        return iterator.hasNext();
+    }
+
+    @Override
+    public T getNext(long time) {
+        if (closed) {
+            throw new IllegalStateException("Data source is already closed.");
+        }
+        return rowMapper.map(iterator.next());
+    }
+
+    @Override
+    public void close() throws Exception {
+        closed = true;
+    }
+
+    /**
+     * Maps single CVS row represented by {@link CSVRecord} to result object.
+     *
+     * @param <T> Type of result object.
+     */
+    public interface RowMapper<T> {
+
+        /**
+         * Returns mapped object of {@code <T>} type.
+         *
+         * @param record record to be mapped.
+         * @return mapped object of {@code <T>} type.
+         */
+        T map(CSVRecord record);
```

Comment:
How should a mapper instance handle a case where the `record` cannot be mapped to the target type?